platforms/core-runtime/wrapper-shared/src/main/java/org/gradle/wrapper/PropertiesFileHandler.java

```diff
@@ -15,21 +15,57 @@
  */
 package org.gradle.wrapper;
 
+import org.gradle.util.internal.ArgumentsSplitter;
+
 import java.io.File;
 import java.io.FileInputStream;
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 import java.util.Properties;
 
-public class SystemPropertiesHandler {
-    static final String SYSTEM_PROP_PREFIX = "systemProp.";
+public class PropertiesFileHandler {
+    private static final String SYSTEM_PROP_PREFIX = "systemProp.";
+    private static final String JVMARGS_PROP_KEY = "org.gradle.jvmargs";
 
     public static Map<String, String> getSystemProperties(File propertiesFile) {
-        Map<String, String> propertyMap = new HashMap<String, String>();
         if (!propertiesFile.isFile()) {
-            return propertyMap;
+            return Collections.emptyMap();
         }
+        Properties properties = loadProperties(propertiesFile);
+        Map<String, String> systemProperties = new HashMap<String, String>();
+        for (Object argument : properties.keySet()) {
+            if (argument.toString().startsWith(SYSTEM_PROP_PREFIX)) {
+                String key = argument.toString().substring(SYSTEM_PROP_PREFIX.length());
+                if (key.length() > 0) {
+                    systemProperties.put(key, properties.get(argument).toString());
+                }
+            }
+        }
+        return Collections.unmodifiableMap(systemProperties);
```

Comment:
The logic here is unchanged; I only extracted the `loadProperties` to reuse it in `getJvmArgs()`.